pkg/acpi/sdt.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package acpi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+)
+
+var (
+	sdtLen = 36
+)
+
+// SDT contains information about tables. It does not differentiate 32- vs 64-bit
+// tables.
+type SDT struct {
+	// Table is the SDT itself.
+	Table
+
+	// Addrs is the array of physical addresses in the SDT.
+	Addrs []int64
+
+	// Base is the SDT base, used to generate a new SDT for, e.g, coreboot.
+	Base int64
+}
+
+var _ = Table(&SDT{})
+
+// NewSDTAddr returns an SDT, given an address.
+func NewSDTAddr(addr int64) (*SDT, error) {
+	t, err := ReadRawTable(addr)
+	if err != nil {
+		return nil, fmt.Errorf("can not read SDT at %#x", addr)
+	}
+	Debug("NewSDTAddr: %s %#x", String(t), t.TableData())
+	return NewSDT(t, addr)
+}
+
+// NewSDT returns an SDT, given a Table
+func NewSDT(t Table, addr int64) (*SDT, error) {
+	s := &SDT{
+		Table: t,
+		Base:  addr,
+	}
+	r := bytes.NewReader(t.TableData())
+	x := true
```

Comment:
can you rename x to something descriptive, like "is64bit"

Reply:
done, FWIW, the x to mean meant "X as in XSDT" but agreed a nicer name is nicer.